### DISCUSSION TITLE
Permission denied (publickey)

### DISCUSSION THREAD

**Discussion by jrndberg** [semaphoreui/semaphore#1744] [Q&A]
```
TASK [Gathering Facts] *********************************************************
fatal: [myhost]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: user@myhost: Permission denied (publickey).", "unreachable": true}
```
when using a Ed25519 key.

**Comment by schiaro98** [comment]
Please describe the environment and how you setup it

**Comment by jrndberg** [comment]
> Please describe the environment and how you setup it

Hi, this is a docker container. I am not exactly sure what you mean. The host running the container is able to SSH to the hosts in question just fine using this SSH key, but when added the key to my keystore I get the above mentioned error

**Comment by jrndberg** [comment]
Update:

I crated a key using RSA sha-2048 and now it seems to work, so I believe there is an issue with using the GitLab keys

**Comment by andreas-marschke** [comment]
SSH client and keys configuration is container and user inside container local. The host for the running container is absolutely irrelevant to verify if it's a bug in semaphore or not. 

Exec into the container and verify the semaphore user has 1) the keys and 2) the ability to use keys in the cipher you describe.

Thanks.
